Source/WebCore/platform/mediastream/mac/MockRealtimeVideoSourceMac.mm

```diff
@@ -118,6 +118,12 @@
     dispatchVideoFrameToObservers(*videoFrame, metadata);
 }
 
+void MockRealtimeVideoSourceMac::endProducingData()
+{
+    MockRealtimeVideoSource::stopProducingData();
+    stopRunLoop();
```

Comment:
Could you add this in the GStreamer mock source too? I suspect we have the same issue there.